### ISSUE TITLE
State mismatch when login via oidc

### ISSUE THREAD

**Issue by wayilau** [goharbor/harbor#9384]
As i give a oidc provider using hydra. I implement login & consent, when i login via oidc, return bad request. logs:
[/core/controllers/oidc.go:81]: State mismatch, in session: %!s(<nil>), in url: 5jr5r0bpypsj3ft9fs280mw4lnei4pg2

It seems that there is not state value in session. I don't know why.

**Comment by wayilau** [comment]
![image](https://user-images.githubusercontent.com/3197838/66700851-34b0fd80-ed28-11e9-91a9-de9357a8e274.png)
As we can see the state generated first. but get nil after a monment.

**Comment by wayilau** [comment]
@xaleeks

**Comment by reasonerjt** [comment]
The state was stored in session, is it possible due to your setup when you are redirected back to Harbor the session is lost?

**Comment by wayilau** [comment]
yes, thanks for your reply, i am checking the redirected settings.

**Comment by wayilau** [comment]
resolved, thanks a lot ! @reasonerjt

**Comment by avaussant** [comment]
Just one add on for this ticket we have the same troubles on Harbor UI with external redis ha.
 
In my situation the redis was deploy from the stable/redis-ha chart and for me it was logical to activate the option stickyBalancing for haporxy. But this option induces the same issue multiple disconnections

the fix was to add the value to false (initial state) 
and 
add nginx.ingress.kubernetes.io/affinity-mode: "persistent" for the ingress controller, we have 3 replicas (3azs) for all critical components

**Comment by Vad1mo** [comment]
@wayilau how did you resolve that issue?

**Comment by MrD2** [comment]
How did you solve this @wayilau?

What was the origin of the problem that you had?

**Comment by orobourosfkt** [comment]
@wayilau @Vad1mo @dioguerra i have the same problem on a fresh install...

Do you have any solution to resolve that issue ?
